### PR TITLE
add tests with config_create from db

### PR DESCRIPTION
**Pull request by Milerius**

### CODE REVIEW

File: service/protocol.hpp [hunk 1]
```diff
@@ -19,7 +19,8 @@ namespace raven
     unknown_id,
     unknown_key,
     unknown_setting,
-    unknown_alias
+    unknown_alias,
+    db_error = -1
```
Comment (Gaspard--) [inline service/protocol.hpp]: Any reason for the negative id?

Comment (Gaspard--) [reply service/protocol.hpp]: I think you're doing this in order to be able to return it as a config id. I'd question the fact that it is a request state in that case.

Comment (Milerius) [reply service/protocol.hpp]: Perfectly right, negative value is only for this case

Comment (Milerius) [reply service/protocol.hpp]: In case of error we get an request_state set to error_db otherwise the column_id used as a CONFIG_KEY

Comment (Gaspard--) [reply service/protocol.hpp]: I think it's minor anyway, merging as is.

**Pull request merged**